modules/economy-system/events/ready.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Wait until the bot is ready
+ * @param {Client} client Client
+ * @returns {Promise}
+ * @private
+ */
+async function checkReady(client) {
+    return new Promise(async (resolve) => {
+
+        /**
+         * Check if the bot is ready
+         * @returns {Promise<void>}
+         * @private
+         */
+        function check() {
+            if (client.botReadyAt) return resolve();
+            else {
+                setTimeout(check, 1000);
+            }
+        }
+        setTimeout(check, 10000);
+    });
+}
```

Comment:
but... why? Why not use the `botReady` event?

Reply:
I use the ready event, because otherwise the cooldowns gets reseted every time when the config gets reloaded

Reply:
Okay... there has to be a better way to do this then looping indefinitely, please don't close this (just) yet, I'm gonna check some stuff an/or implement something more efficient than this ^^

Reply:
Okay, there are two options to fix this:
1. Add `client.lastReloadAt` paramenter (Date) which gets set at every reload and can be used to determine, if this is the first restart.
2. Instead of using timeouts, store the expiring timestamp in the database and check once a user tries to work if the cooldown passed already. 

I think the second option is the better one, as with this method everyone has to wait for their countdown to expire and reduces database-access (you don't have to delete anything). It also is way more perfroment then the first one, as you don't need to create a timeout on every user request.  

As always, feel free to share your thoughts if you disagree. I am also willing to implement the secound option myself if you currently don't have enough time to invest in this project ^^